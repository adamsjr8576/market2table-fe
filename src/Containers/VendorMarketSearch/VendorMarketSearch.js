import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import MarketListCard from '../MarketListCard/MarketListCard';
import './VendorMarketSearch.scss';
import { getMarketsByZip } from '../../apiCalls';
import { addZipCode, addMarkets } from '../../actions';

export const VendorMarketSearch = ({ marketsLinked, setMarketsLinked }) => {
  const [hasError, setHasError] = useState(false);
  const [zipCodeInput, setZipCodeInput] = useState('');
  const {zipCode, markets, selectedVendor} = useSelector(state => state);
  const dispatch = useDispatch();

  const handleZipCodeSubmit = () => {
    if (zipCodeInput.length === 5) {
      getMarketsByZip(zipCodeInput)
        .then(markets => {
          dispatch(addMarkets(markets));
        })
      .catch(error => console.log(error))
      zipCodeText = zipCodeInput;
      dispatch(addZipCode(zipCodeInput));
      setZipCodeInput('');
    } else {
      setHasError(true);
      setZipCodeInput('');
    }
  }

  let zipCodeText = 'Please enter zip code';
  if (zipCode.length === 5) {
    zipCodeText = zipCode;
  }


  let marketsToDisplay = (<p>Please select a zip code to display markets</p>);

  if (markets.length) {
    const filteredMarkets = markets.filter(market => {
      const filter = marketsLinked.filter(marketLinked => market.name === marketLinked.name);
      return !filter.length;
    });
    marketsToDisplay = filteredMarkets.map(market => {
      return (
        <MarketListCard vendorId={selectedVendor.id} name={market.name} id={market.id} key={market.id} setMarketsLinked={setMarketsLinked} marketsLinked={marketsLinked} vendorDescription={selectedVendor.description} />
      )
    });
  }

  return (
    <section className='vendor-market-search-section'>
      <h3 className='vendor-market-search-title'>Search For Markets</h3>
      <form className='vendor-market-search-form'>
        <p className='vendor-market-search-par'>Zip Code: {zipCodeText}</p>
        <div className='vendor-market-search-container'>
          <input value={zipCodeInput} onChange={(event) => {setZipCodeInput(event.target.value); setHasError(false)}} className='vendor-market-input' type='text' type="number" placeholder="zip code..." id="zipCode" name="zipCode" />
          <button onClick={handleZipCodeSubmit} className='vendor-market-search-button' type='button'>Find!</button>
          <div className='error-message-search-container'>
            <p hidden={!hasError} className='error-message'>Error: Please enter a 5 digit zip code</p>
          </div>
        </div>
      </form>
      <section className='vendor-market-display-container'>
        {marketsToDisplay}
      </section>
    </section>
  )
}

export default VendorMarketSearch;
