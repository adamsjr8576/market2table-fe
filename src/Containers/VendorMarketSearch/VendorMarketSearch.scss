.vendor-market-search-section {
  width: 45%;
  height: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  background-color: white;
  border: 3px solid black;
  border-radius: 20px;
}

.vendor-market-search-title {
  font-size: 2em;
  margin: 10px 0;
  text-decoration: underline;
}

.vendor-market-search-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 150px;
  width: 100%;
}

.vendor-market-search-label-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
}

.vendor-market-search-label {
  font-size: 1.5em;
  width: 35%;
}

.vendor-market-search-par {
  font-size: 1.5em;
  width: 65%;
  margin: 0;
}

.vendor-market-search-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
}

.vendor-market-input {
  font-size: 1.5em;
  font-size: 1.5em;
  border-radius: 20px;
  outline: 0;
  border: 0;
  padding: 7px;
  text-align: center;
  appearance: none;
  border: 2px solid black;
}

.vendor-market-search-button {
  width: 20%;
  font-size: 1.5em;
  border-radius: 20px;
  padding: 5px;
  min-width: 100px;
  cursor: pointer;
  outline: 0;
}

.error-message-search-container {
  width: 20%;
}
