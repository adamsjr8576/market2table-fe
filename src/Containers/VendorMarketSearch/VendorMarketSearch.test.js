import React from 'react';
import { shallow } from 'enzyme';
import { VendorMarketSearch } from './VendorMarketSearch';

const mockState = {
  selectedVendor: {name: 'Rolling in the dough', description: 'just a little bakery rolling in some dough', image_link: 'rollthatdough.png'},
  zipCode: '80401',
  markets: [{name: 'Friendly Famrs', address: '1234 Hillsbury Lane', schedule: 'M-F ALL DAY', google_link: 'google.com', id:'1923782'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('VendorMarketSearch', () => {

  let wrapper;
  let mockAddZipCode = jest.fn();
  let mockAddMarkets = jest.fn();
  let mockMarketsLinked;
  let mockSetMarketsLinked = jest.fn();

  it('should match the snapshot', () => {
    mockMarketsLinked = [{name: 'Friendly Famrs', address: '1234 Hillsbury Lane', schedule: 'M-F ALL DAY', google_link: 'google.com', id:'1923782'}, {name: 'farm farm', address: '1234 Hillsbury Lane', schedule: 'M-F ALL DAY', google_link: 'google.com', id:'1923782'}];
    wrapper = shallow(<VendorMarketSearch
      addZipCode={mockAddZipCode}
      addMarkets={mockAddMarkets}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with no match', () => {
    mockMarketsLinked = [];
    wrapper = shallow(<VendorMarketSearch
      addZipCode={mockAddZipCode}
      addMarkets={mockAddMarkets}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
    />);

    expect(wrapper).toMatchSnapshot();
  });
  it("Should have a default state on zipCode input of ''", () => {
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('');
  });

  it("Should invoke onChange and update state if value is less than 6 on zip code input", () => {
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80401'}});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('80401');
  });

  it("Should onClick invoke handleZipCodeSubmit and reset zip code input to '' if input.length is 5", () => {
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80401'}});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('80401');
    wrapper.find('.vendor-market-search-button').simulate('click', {preventDefault: jest.fn()});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('');
  });

  it("Should onClick invoke handleZipCodeSubmit and reset zip code input to '' if input.length is 5", () => {
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80'}});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('80');
    wrapper.find('.vendor-market-search-button').simulate('click', {preventDefault: jest.fn()});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('');
    expect(wrapper.find('.error-message').prop('hidden')).toEqual(false);
  });

  it("should have a default value of false for hasError", () => {
    expect(wrapper.find('.error-message').prop('hidden')).toEqual(false);
  });

  it("Should onClick invoke handleZipCodeSubmit and reset zip code input to '' if input.length is 5", () => {
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80401'}});
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('80401');
    wrapper.find('.vendor-market-search-button').simulate('click');
    expect(wrapper.find('.vendor-market-input').prop('value')).toEqual('');
  });

  it("should have a value of true for hasError when not length of 5 and then set to false on input change", () => {
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80'}});
    wrapper.find('.vendor-market-search-button').simulate('click');
    expect(wrapper.find('.error-message').prop('hidden')).toEqual(false);
    wrapper.find('.vendor-market-input').simulate('change', {target: {value: '80401'}});
    expect(wrapper.find('.error-message').prop('hidden')).toEqual(true);
  });

});
