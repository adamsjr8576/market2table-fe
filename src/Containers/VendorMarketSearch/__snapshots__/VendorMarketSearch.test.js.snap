// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorMarketSearch should match the snapshot 1`] = `
<section
  className="vendor-market-search-section"
>
  <h3
    className="vendor-market-search-title"
  >
    Search For Markets
  </h3>
  <form
    className="vendor-market-search-form"
  >
    <p
      className="vendor-market-search-par"
    >
      Zip Code: 
      80401
    </p>
    <div
      className="vendor-market-search-container"
    >
      <input
        className="vendor-market-input"
        id="zipCode"
        name="zipCode"
        onChange={[Function]}
        placeholder="zip code..."
        type="number"
        value=""
      />
      <button
        className="vendor-market-search-button"
        onClick={[Function]}
        type="button"
      >
        Find!
      </button>
      <div
        className="error-message-search-container"
      >
        <p
          className="error-message"
          hidden={true}
        >
          Error: Please enter a 5 digit zip code
        </p>
      </div>
    </div>
  </form>
  <section
    className="vendor-market-display-container"
  />
</section>
`;

exports[`VendorMarketSearch should match the snapshot with no match 1`] = `
<section
  className="vendor-market-search-section"
>
  <h3
    className="vendor-market-search-title"
  >
    Search For Markets
  </h3>
  <form
    className="vendor-market-search-form"
  >
    <p
      className="vendor-market-search-par"
    >
      Zip Code: 
      80401
    </p>
    <div
      className="vendor-market-search-container"
    >
      <input
        className="vendor-market-input"
        id="zipCode"
        name="zipCode"
        onChange={[Function]}
        placeholder="zip code..."
        type="number"
        value=""
      />
      <button
        className="vendor-market-search-button"
        onClick={[Function]}
        type="button"
      >
        Find!
      </button>
      <div
        className="error-message-search-container"
      >
        <p
          className="error-message"
          hidden={true}
        >
          Error: Please enter a 5 digit zip code
        </p>
      </div>
    </div>
  </form>
  <section
    className="vendor-market-display-container"
  >
    <MarketListCard
      id="1923782"
      key="1923782"
      marketsLinked={Array []}
      name="Friendly Famrs"
      setMarketsLinked={[MockFunction]}
      vendorDescription="just a little bakery rolling in some dough"
    />
  </section>
</section>
`;
